src/libs/antares/study/scenario-builder/hydroFinalLevelsData.h

```diff
@@ -0,0 +1,117 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#ifndef __LIBS_STUDY_SCENARIO_BUILDER_DATA_HYDRO_FINAL_LEVELS_H__
+#define __LIBS_STUDY_SCENARIO_BUILDER_DATA_HYDRO_FINAL_LEVELS_H__
+
+#include "scBuilderDataInterface.h"
+
+namespace Antares
+{
+namespace Data
+{
+namespace ScenarioBuilder
+{
+/*!
+** \brief Rules for hydro final levels, for all years and areas
+*/
+class hydroFinalLevelsData final : public dataInterface
+// TODO CR25: can i inherit methods from hydroLevelsData (must delete final in its declaration?!) 
+{
+public:
+    //! Matrix
+    using MatrixType = Matrix<double>;
+
+public:
```

Comment:
Yes : as already pointed, there is likely a lot of duplication code in classes  **hydroFinalLevelsData** and **hydroLevelsData**. 
The only difference **seems** to be in printing the level name in **scenariobuilder.dat** (see member function `saveToINIFile(...)` and the line : `prefix += "hl,";`)

Reply:
yes, see https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1175#discussion_r1123000160